Gems/ROS2/Code/Include/ROS2/Camera/CameraSensorEditorRequestBus.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Component/EntityId.h>
+#include <AzCore/EBus/EBus.h>
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/std/string/string.h>
+
+#include <sensor_msgs/msg/image.hpp>
+
+#include "ImagePostprocessingFunction.h"
+
+namespace ROS2
+{
+    //! Interface class that allows for communication with the Camera Sensor Editor Component.
+    class CameraSensorEditorRequests : public AZ::EBusTraits
+    {
+    public:
+        static constexpr AZ::EBusAddressPolicy AddressPolicy = AZ::EBusAddressPolicy::ById;
+        using BusIdType = AZ::EntityId;
+
+        //! Register a depth post-processing function.
+        //! Only one post processing can be registered, the new one overrides the former.
+        //! @param postProcessingFunction function which will be called synchronously after rendering pipeline but before publishing.
+        virtual void RegisterDepthPostProcessing(const Camera::ImagePostProcessingFn& postProcessingFunction) = 0;
+
+        //! Register a color post-processing function.
+        //! Only one post processing can be registered, the new one overrides the former.
+        //! @param postProcessingFunction function which will be called synchronously after rendering pipeline but before publishing.
+        virtual void RegisterColorPostProcessing(const Camera::ImagePostProcessingFn& postProcessingFunction) = 0;
+
+    protected:
+        virtual ~CameraSensorEditorRequests() = default;
```

Comment:
nit: You can drop the `virtual` here as we never intend to delete the object through this request interface (and having the destructor in the protected section guards against improper use just in case 🙂)